modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/SourceContext.java

```diff
@@ -138,19 +152,47 @@ public void setWriter(Pipe writer) {
 
     public static void create(NHttpConnection conn, ProtocolState state,
                               SourceConfiguration configuration) {
-        SourceContext info = new SourceContext(configuration);
-
-        conn.getContext().setAttribute(CONNECTION_INFORMATION, info);
+        SourceContext sourceContext = new SourceContext(configuration);
+        conn.getContext().setAttribute(CONNECTION_INFORMATION, sourceContext);
+        sourceContext.setState(state);
 
-        info.setState(state);
+        if (sourceContext.getSourceConfiguration().isCorrelationLoggingEnabled()) {
+            sourceContext.updateLastStateUpdatedTime();
+        }
     }
 
     public static void updateState(NHttpConnection conn, ProtocolState state) {
-        SourceContext info = (SourceContext)
+        SourceContext sourceContext = (SourceContext)
                 conn.getContext().getAttribute(CONNECTION_INFORMATION);
 
-        if (info != null) {
-            info.setState(state);
+        if (sourceContext != null) {
+            if (sourceContext.getSourceConfiguration().isCorrelationLoggingEnabled()) {
+                long lastStateUpdateTime = sourceContext.getLastStateUpdatedTime();
+                String url = "", method = "";
+                if (sourceContext.getRequest() != null) {
+                    url = sourceContext.getRequest().getUri();
+                    method = sourceContext.getRequest().getMethod();
+                } else {
+                    HttpRequest httpRequest = conn.getHttpRequest();
+
```

Comment:
Shall we remove unnecessary spaces here?

Reply:
fixed in https://github.com/sachithKay/wso2-synapse/commit/3373b68eeae9c06550355732e1f7a4fbc6c42153

Reply:
Fixed : https://github.com/wso2/wso2-synapse/blob/3af0f7ee38db87e60a0b15095c35672d188cf815/modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/SourceContext.java#L175